cpp/Platform.Equality.Tests/Platform.Equality.Tests.cpp

```diff
@@ -0,0 +1,82 @@
+#include <gtest/gtest.h>
+#include <Platform.Equality.h>
+
+using namespace Platform::Equality;
+
+namespace PlatformEqualityTests
+{
+    TEST(EqualityTest, BaseAnyEqual)
+    {
+        {
+            int a = 0x228;
+            int b = 0x1337;
+
+            ASSERT_TRUE(std::any(a) != std::any(b));
```

Comment:
better to use `ASSERT_NE(lhs, rhs);` - it writes more debug information in case of an error